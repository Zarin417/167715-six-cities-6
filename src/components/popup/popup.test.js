import React from "react";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import Popup from "./popup";

const mockStore = configureStore({});

describe(`Popup should render correctly`, () => {
  it(`Popup should show 'Data loading error' message`, () => {
    const store = mockStore({
      OFFER: {errorCode: null}
    });

    const {container} = render(
        <Provider store={store}>
          <Popup />
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`Popup should show 'Wrong email or password' message`, () => {
    const store = mockStore({
      OFFER: {errorCode: 400}
    });

    const {container} = render(
        <Provider store={store}>
          <Popup />
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});


