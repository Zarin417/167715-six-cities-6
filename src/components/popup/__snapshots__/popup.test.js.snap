// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should render correctly Popup should show 'Data loading error' message 1`] = `
<div>
  <section
    class="error"
  >
    <div
      class="error__inner"
    >
      <h1
        class="error__title"
      >
        Data loading error
      </h1>
      <button
        class="error__button button"
      >
        Close
      </button>
    </div>
  </section>
</div>
`;

exports[`Popup should render correctly Popup should show 'Wrong email or password' message 1`] = `
<div>
  <section
    class="error"
  >
    <div
      class="error__inner"
    >
      <h1
        class="error__title"
      >
        Data loading error
      </h1>
      <button
        class="error__button button"
      >
        Close
      </button>
    </div>
  </section>
</div>
`;
